### PR TITLE
feat: allow new telegraf input plugins and update toml

### PR DESCRIPTION
**Pull request by TCL735**
Closes idpe 11656

Matching pull request for idpe pull request 11740

- allow additional input plugins to be valid and do not return an error if a plugin is an input and not on the list
- update the toml configuration code for agent, output, and input
- update test assertions

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [x] Feature flagged (if modified API)
- [x] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Review by glinton — APPROVED** [review]

File: telegraf/plugins/plugins.go [hunk 1]
```diff
@@ -841,8 +863,8 @@ var availableInputs = `{
     {
       "type": "input",
       "name": "puppetagent",
-      "description": "Reads last_run_summary.yaml file and converts to measurements",
-      "config": "# Reads last_run_summary.yaml file and converts to measurements\n[[inputs.puppetagent]]\n  # alias=\"puppetagent\"\n  ## Location of puppet last run summary file\n  location = \"/var/lib/puppet/state/last_run_summary.yaml\"\n\n"
+      "description": "Reads last_run_summary.yaml file and converts to measurments",
```
Comment (wbaker85) [inline telegraf/plugins/plugins.go:866]: Typo?

File: telegraf/plugins/plugins.go [hunk 2]
```diff
@@ -1177,8 +1199,8 @@ var availableInputs = `{
     {
       "type": "input",
       "name": "marklogic",
-      "description": "Retrieves information on a specific host in a MarkLogic Cluster",
-      "config": "# Retrieves information on a specific host in a MarkLogic Cluster\n[[inputs.marklogic]]\n  # alias=\"marklogic\"\n  ## Base URL of the MarkLogic HTTP Server.\n  url = \"http://localhost:8002\"\n\n  ## List of specific hostnames to retrieve information. At least (1) required.\n  # hosts = [\"hostname1\", \"hostname2\"]\n\n  ## Using HTTP Basic Authentication. Management API requires 'manage-user' role privileges\n  # username = \"myuser\"\n  # password = \"mypassword\"\n\n  ## Optional TLS Config\n  # tls_ca = \"/etc/telegraf/ca.pem\"\n  # tls_cert = \"/etc/telegraf/cert.pem\"\n  # tls_key = \"/etc/telegraf/key.pem\"\n  ## Use TLS but skip chain \u0026 host verification\n  # insecure_skip_verify = false\n\n"
+      "description": "Retrives information on a specific host in a MarkLogic Cluster",
```
Comment (wbaker85) [inline telegraf/plugins/plugins.go:1202]: Typo?

File: telegraf/plugins/outputs/outputs_test.go [hunk 3]
```diff
@@ -65,22 +122,45 @@ func TestTOML(t *testing.T) {
...
+  ## Each data format has its own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "influx"
 `,
 				&InfluxDBV2{
 					URLs: []string{
-						"http://192.168.1.10:8086",
-						"http://192.168.1.11:8086",
+						"http://192.168.1.10:9999",
```
Comment (wbaker85) [inline telegraf/plugins/outputs/outputs_test.go:151]: What is the significance of changing these to 9999? The default port for OSS is 8086 - does this show up anywhere but the tests?

File: telegraf/plugins/plugins.go (hunk 1 above)
Comment (TCL735) [reply telegraf/plugins/plugins.go:866]: Yes. Fixed.

File: telegraf/plugins/plugins.go (hunk 2 above)
Comment (TCL735) [reply telegraf/plugins/plugins.go:1202]: Yes, fixed.

File: telegraf/plugins/outputs/outputs_test.go (hunk 3 above)
Comment (TCL735) [reply telegraf/plugins/outputs/outputs_test.go:151]: No significance. Just for the tests. I think the example in the comments of the Telegraf plugins use 9999 and that's where it came from.

Comment (wbaker85) [reply telegraf/plugins/outputs/outputs_test.go:151]: 10-4, seems fine then!

**Review by wbaker85 — APPROVED** [review]
LGTM

**Pull request merged**